feat/product card component

ICarlosMarinho opened the pull request:
Implementa componente ProductCard, módulo utils e adiciona alguns estilos globais.

ICarlosMarinho committed:
feat: adiciona tipo Product e estilos globais

ICarlosMarinho committed:
feat: implementa módulo utils

Implementa módulo que deverá conter funções utilitárias, bem como os testes
para cada função a medida que forem sendo adicionadas.

ICarlosMarinho committed:
feat: implementa componente ProductCard

Implementa componente que irá exibir dados resumidos de um produto,
bem como os testes do componente e os estilos.

michellehorn commented on src/components/ProductCard/test.tsx:
```diff
@@ -0,0 +1,63 @@
+    const button = screen.getByText(text);
+
+    expect(button).toBeInTheDocument();
+  });
```
poderias criar apenas um teste `it("should render product props")`, renderizar o componente uma vez e fazer as assercoes. Alem disso, para validar items como h3, h4, vc pode utilizar o `getByRole()`. Por exemplo, para testar o texto de um h3, ficaria: 
```
const headingh3 = getByRole('heading', { level: 3 })
expect(headingh3).toBeInTheDocument();
```
Já as tags `img`, vc pode buscar pelo alt, utilizando o `getByAltText()`, aí ja valida a presença do atributo alt. Além disso, no expect pode validar o atributo src, por ex: `expect(image.src).toContain('url');`

dessa forma, vc valida que o texto existe e tambem que esta utilizando a tag correta.

ICarlosMarinho committed:
test: refatora e adiciona novos testes

Refatora os testes, de forma que agora um único teste verifica se
as propriedades estão sendo renderizadas conforme a implementação.

Adiciona teste que verifica se ao clicar no link, o usuário é direcionado
para a página de detalhes do produto corretamente.

michellehorn reviewed (approved):
✨🥳  LGTM  🥳✨
